test/integration/targets/unarchive/tasks/main.yml

```
@@ -659,3 +659,50 @@
 
 - name: remove our tar.gz unarchive destination
   file: path={{ remote_tmp_dir }}/test-unarchive-tar-gz state=absent
+
+# Test unarchiving as test user with primary group root
+- name: Create test user with root primary group
+  user:
+    name: unarchivetest1
+    group: root
+
+- name: Test unarchiving twice as test user
+  block:
+  - name: create our zip unarchive destination as test user
+    file: path={{remote_tmp_dir}}/test-unarchive-zip state=directory
```

Please use multi-line YAML for new tests.
```suggestion
    file: 
      path: "{{ remote_tmp_dir }}/test-unarchive-zip"
      state: directory
```

---

I will update my branch and rebase it with this suggestion.